### PR TITLE
send 403 code when printing error page after ForbiddenException

### PR DESCRIPTION
**Pull request by individual-it**
## Description
1. allow the `printErrorPage` function in the template to set the status code header
2. when the access to a (public) file is forbidden send a 403 status code

## Related Issue
https://github.com/owncloud/firewall/issues/322

## Motivation and Context
if the access to a file is forbidden the HTTP status code should be 403 not 200

## How Has This Been Tested?
- made public share of a file
-  blocked the access by file firewall
- accessed the public link

## Screenshots (if appropriate):
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by phil-davis — APPROVED** [review]
This looks like "a good thing".
As a separate issue, for LockedException and the general Exception, printErrorPage should probably also return some decent HTTP "I could not do it" status.

File: lib/private/legacy/template.php [hunk 1]
```diff
@@ -319,7 +319,7 @@ public static function printGuestPage( $application, $name, $parameters = []) {
 		* @param string $error_msg The error message to show
 		* @param string $hint An optional hint message - needs to be properly escaped
 		*/
-	public static function printErrorPage( $error_msg, $hint = '' ) {
+	public static function printErrorPage( $error_msg, $hint = '', $status = null ) {
```
Comment (PVince81) [inline lib/private/legacy/template.php]: phpdoc missing. in general I usually expect the status to just be a number.

maybe the header could simply be set before calling `printErrorPage` ?

**Comment by individual-it** [comment]
@PVince81 pushed a version where the header is set in the `files.php` but don't like that too much. Feels better if the template is responsible for all outputs to the browser

**Comment by PVince81** [comment]
@individual-it I don't like that we're passing the full string where in many other APIs we just pass the status code. Maybe then add two arguments "$statusCode" and "$statusText" ?

**Comment by individual-it** [comment]
@PVince81 actually we don't need the message at all, because we don't set any custom one. So what about setting the code only and using `http_response_code` ?

**Comment by PVince81** [comment]
@individual-it if the http status code value is enough then let's just use that

**Comment by phil-davis** [comment]
Backport to stable10 in #28398 
will be needed do future tests will run OK in stable10

**Pull request merged**